frontend/src/user/components/ChangePasswordForm.tsx

```diff
@@ -0,0 +1,93 @@
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useMutation } from "@tanstack/react-query";
+
+import { Axios } from "src/services";
+import {
+  ErrorAlert,
+  SuccessAlert,
+  TextInput,
+  PrimaryButton,
+} from "src/components";
+import {
+  ApiResponse,
+  ChangePasswordInfo,
+  ChangePasswordInfoSchema,
+} from "src/user/types";
+
+const ChangePasswordForm = () => {
+  // form setup
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm<ChangePasswordInfo>({
+    resolver: zodResolver(ChangePasswordInfoSchema),
+  });
+
+  const changePasswordMutation = useMutation(
+    (params: ChangePasswordInfo) =>
+      Axios.post<ApiResponse>(`/users/change-password`, params).then(
+        (res) => res.data
+      ),
+    {
+      onError: (error) => {
+        console.log({ error });
+      },
+    }
+  );
+
+  // submit function
+  const handleResetPassword = async (credentials: ChangePasswordInfo) => {
+    changePasswordMutation.mutate(credentials);
+    reset();
```

Comment:
let's move the `reset()` function as an `onSuccess` prop in `useMutations` (only resets the form if the POST http query is sucessful), can we also store the `changePasswordMutation` in the `AuthStore`?